### DISCUSSION TITLE
gO for other filetypes

### DISCUSSION THREAD

**Discussion by avidseeker** [vim/vim#16015] [Q&A]
```
							*gO*
gO			Show a filetype-specific, navigable "outline" of the
			current buffer. For example, in a |help| buffer this
			shows the table of contents.

			Currently works in |help| and |:Man| buffers.
```

Is `gO` supported in other file types than help or man? If not, is there a way to set it up (say without a dedicated plugin). For example, if `gO` can be customized to list all headings (which Vim already knows about by `[[` and `]]` navigation).

**Comment by zeertzjq** [comment]
There is no such mapping in Vim. And no, you will need a dedicated plugin to create such a mapping that calls `setloclist()` and `:lopen` to show the TOC.

**Reply by chrisbra** [reply]
We could add the `gO` mapping to call the `:HelpTOC` plugin

**Reply by saccarosium** [reply]
I think it is much better having a more general solution and ideally something that uses the location list since I believe is the more vimy way of doing something like this.
